src/vdaf.rs

```diff
@@ -193,85 +166,78 @@ where
     let num_shares = num_shares as usize;
 
     // Generate the input shares and compute the joint randomness.
-    let mut helper_shares = vec![HelperShare::new(); num_shares - 1];
+    let mut helper_shares = vec![HelperShare::new(suite)?; num_shares - 1];
     let mut leader_input_share = input.as_slice().to_vec();
-    let mut joint_rand_seed = vec![0; Aes128Ctr::STREAM_CIPHER_SEED_SIZE];
+    let mut joint_rand_seed = Key::uninitialized(suite);
     let mut aggregator_id = 1; // ID of the first helper
     for helper in helper_shares.iter_mut() {
-        getrandom(&mut helper.blind)?;
-        getrandom(&mut helper.input_share)?;
-
-        let mut hasher = blake3::Hasher::new();
-        hasher.update(&[aggregator_id]);
-        hasher.update(&helper.blind);
-        let prng: Prng<F, Aes128Ctr> = Prng::new_with_seed(&helper.input_share)?;
+        let mut deriver = KeyDeriver::from_key(&helper.blind);
+        deriver.update(&[aggregator_id]);
+        let prng: Prng<F> = Prng::from_expander(KeyExpander::from_key(&helper.input_share));
         for (x, y) in leader_input_share.iter_mut().zip(prng).take(input_len) {
             *x -= y;
-            hasher.update(&y.into());
+            deriver.update(&y.into());
         }
 
-        helper
-            .joint_rand_seed_hint
-            .copy_from_slice(hasher.finalize().as_bytes());
+        helper.joint_rand_seed_hint = deriver.finish();
         for (x, y) in joint_rand_seed
+            .as_mut_slice()
             .iter_mut()
-            .zip(helper.joint_rand_seed_hint.iter())
+            .zip(helper.joint_rand_seed_hint.as_slice().iter())
         {
             *x ^= y;
         }
 
         aggregator_id += 1; // ID of the next helper
     }
 
-    let mut leader_blind = vec![0; Aes128Ctr::STREAM_CIPHER_SEED_SIZE];
-    getrandom(&mut leader_blind)?;
+    let leader_blind = Key::generate(suite)?;
 
-    let mut hasher = blake3::Hasher::new();
-    hasher.update(&[0]); // ID of the leader
-    hasher.update(&leader_blind);
+    let mut deriver = KeyDeriver::from_key(&leader_blind);
+    deriver.update(&[0]); // ID of the leader
     for x in leader_input_share.iter() {
-        hasher.update(&(*x).into());
+        deriver.update(&(*x).into());
     }
 
-    let mut leader_joint_rand_seed_hint = vec![0; Aes128Ctr::STREAM_CIPHER_SEED_SIZE];
-    leader_joint_rand_seed_hint.copy_from_slice(hasher.finalize().as_bytes());
+    let mut leader_joint_rand_seed_hint = deriver.finish();
```

Comment:
Yes because its borrowed as mutable below:
```
$ cargo build
   Compiling prio v0.4.0 (/home/chris/github.com/abetterinternet/libprio-rs)
error[E0596]: cannot borrow `leader_joint_rand_seed_hint` as mutable, as it is not declared as mutable
   --> src/vdaf.rs:238:19
    |
203 |     let leader_joint_rand_seed_hint = deriver.finish();
    |         --------------------------- help: consider changing this to be mutable: `mut leader_joint_rand_seed_hint`
...
238 |     for (x, y) in leader_joint_rand_seed_hint
    |                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^ cannot borrow as mutable

For more information about this error, try `rustc --explain E0596`.
error: could not compile `prio` due to previous error
```